doc/specification.md

```diff
@@ -35,62 +35,126 @@ TLS
 
 # 4 Service Definition
 
-The gRIBI service is defined as a single gRPC service, with three RPCs:
+The gRIBI service is a single gRPC service currently defined in [`gribi.proto`](https://github.com/openconfig/gribi/blob/master/v1/proto/service/gribi.proto). It includes three RPCs:
 
- * `Modify` - a bidirectional stream RPC which is used by the client to inject
-   routing entries to the server - each part of an individual operation. The 
-   server responds asynchronously to these operations with an acknowledgement
-   mode, based on the operating mode of the RPC.
+ * `Modify` - used by the clients to modify the device's RIB.
  * `Get` - a server streaming RPC which can be used by a client to retrieve the
    current set of installed gRIBI entries.
  * `Flush` - a unary RPC that is used as a low-complexity means to remove
    entries from a server.
 
-## 4.1 The `Modify` RPC
+IANA has reserved [TCP port 9340](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=9340#Google_Networking) for gRIBI service.
 
-High-level description of `Modify` semantics.
+## 4.1 `Modify`
 
-* `ADD`
-  * repeated `ADD` translates to `MODIFY`
-* `MODIFY`
-  * must fail if referenced object does not exist.
-* `DELETE`
-  * error handling for missing entries
-  * error handling for forward references
+The `Modify` provides a bidirectional streaming RPC for clients to modify the device's RIB. A client sends `ModifyRequest` messages that contains a set of `AFTOperations` to the device. The device process the received requests and responds them asynchronously.
+
+### 4.1.1 Client-Server Session Neogotiation
+
+A gRIBI client is identified by gRPC connection, i.e., if a connection drops and reconnect with the same `election_id` value, it will be considered as another client.
```

Comment:
I think a client is probably identified by a `Modify` RPC. i.e., if I do something like:

```golang

conn, _ := grpc.DialContext(server, ...)
c := gribipb.NewGRIBIClient(conn)

s1, _ := c.Modify(ctx)
s2, _ := c.Modify(ctx)
```

in the same code, then I think that this would be seen as two different clients at the target in the reference server today (reference: https://github.com/openconfig/gribigo/blob/main/server/server.go#L274). Given that gRPC doesn't give us access to the underlying client address (for example), or anything that would let us know that these two different clients are actually the same, I think it is reasonable to say that a gRIBI client is identified by a `Modify` stream.

Reply:
This follows further because at the start of `s1` and `s2` we expect that the election behaviours would need to be specified.